AWS Upload

camcarlton opened the pull request:
Class for pi backend to call AWS APIs and AWS enivornment install.

camcarlton committed:
AWS Upload

Class for pi backend to call AWS APIs and AWS enivornment install.

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+import json
+
+#this class is for telescreen_device activity
+class AWS(object):
```
This class needs a doc string below its definition, see https://google.github.io/styleguide/pyguide.html#Comments

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+import boto3
+import json
+
+#this class is for telescreen_device activity
```
In line comments should only be above tricky parts of the code.

> Complicated operations get a few lines of comments before the operations commence. Non-obvious ones get comments at the end of the line.

I will only comment on this occurrence, but there are many other occurrences below.
see https://google.github.io/styleguide/pyguide.html#Comments

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+		self.session = boto3.Session(profile_name='default')
+	
+	#uploading img to S3
+	def uploadS3(self, img):
```
This method and ones that follow may need doc strings (instead of the inline comments above).

See https://google.github.io/styleguide/pyguide.html#Comments under "Functions and Methods". 

I will let you decide what meets the criteria in the above link required for not having a doc string.

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+		pass
+		
+	#detect if any faces are found within image
+	def detectFace(self, img):  
```
As far as I know we won't be using AWS to detect faces. Is there a reason that we need this one?

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+			print(json.dumps(faceDetail, indent=4, sort_keys=True))	
+	
+	#compare faces with one image to another
+	def compareFaces(self, img):
```
Was this left intentionally blank? Either complete its functionality or remove it. Only methods/functions that are fully functional or contribute to some greater functionality should be in a pull request!

dlowrey commented on sub-projects/device/utils/SetupAWS(windows).ps1:
```diff
@@ -0,0 +1,10 @@
+
+
+
+
```
Remove all blank lines

dlowrey commented on sub-projects/device/utils/SetupAWS(linux).bash:
```diff
@@ -0,0 +1,7 @@
+pip install boto3
+
+pip install awscli
+
```
Remove all blank lines

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+	def uploadS3(self, img):
+		s3 = self.session.resource('s3')
+		s3.meta.client.upload_file(img, 'checkfaces', img)
+		
```
Class methods should be separated by only a single blank line.

See https://google.github.io/styleguide/pyguide.html#Blank_Lines

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+		s3.delete_object(Bucket='checkfaces', Key=img)
+	
+	#get a list of all the known faces from the server database
+	def queryKnownFaces(self):
```
Was this left intentionally blank? Either complete its functionality or remove it. Only methods/functions that are fully functional or contribute to some greater functionality should be in a pull request!

dlowrey commented on sub-projects/device/src/AWSClass.py:
```diff
@@ -0,0 +1,45 @@
+	def compareFaces(self, img):
+		#need to have pictures in the knownfaces bucket
+		
+		
```
Remove any trailing blank lines

dlowrey reviewed (changes requested):
I am not sure how to review the functionality bits because I do not know how to work the AWS side of things, but we can discuss later!

camcarlton committed:
Reformatted code

Went through all the problems that were pointed out from the pull request. Added a test_aws_s3 file for testing the S3 class.
